src/borg/testsuite/archiver/checks.py

```diff
@@ -43,7 +43,7 @@ def cmd_raises_unknown_feature(archiver, args):
 
 
 def pytest_generate_tests(metafunc):
-    # Generate tests for different scenarios: local repository, remote repository, and using the borg binary.
+    # Generate tests for local and remote archivers
     if "archivers" in metafunc.fixturenames:
         metafunc.parametrize("archivers", ["archiver", "remote_archiver"])
```

Comment:
guess that function could have a `kinds` argument and you could use lambda or partial to generate the functions you need from that.

kinds could be given like "local,remote,binary", so it is easier to type.

Reply:
Hmm, I cannot seem to make this work. If I add `kinds` as an argument to `pytest_generate_tests`, then when I import `pytest_generate_tests` into a test module, I cannot call it as `pytest_generate_tests(metafunc, kinds)` because metafunc cannot be given to it. What am I missing here?

Reply:
Guess I'ld rename the function definition to just `generate_archiver_tests`, then import that and then define:
```
pytest_generate_tests = lambda metafunc: generate_archiver_tests(metafunc, kinds="...")
```